Makefile

```diff
@@ -30,8 +30,8 @@ all: build unit-test system-test ubuntu-tests
 # 'all-CI' target is used by the scripts/CI.sh that passes appropriate set of
 # ENV variables (from the jenkins job) to run OS (centos, ubuntu etc) and
 # sandbox specific(vagrant, docker-in-docker)
-all-CI: stop clean start
-	make ssh-build
+all-CI: stop clean
+	make checks run-build start ssh-build
```

Comment:
This is how things were before this PR:
1. stop VMs if they were running (not the case, no VMs were on, safe to do this nonetheless)
2. clean up
3. start the VMs & provision them - takes ~10 minutes
4. run make checks as part of ssh-build - this connects to the first node to run `make checks` and compile everything
5. the tests are run

This is how things would be with this PR:
1. stop the VMs if they were running (same as before)
2. clean up
3. run `make checks` on the host to ensure the code passes the checks (no formatting issues and so on)
4. run `make run-build` to ensure the code compiles
5. start the VMs - takes ~10 minutes
6. run `make ssh-build`, just like step 4 before
7. the tests are run as before

This is an optimization to avoid provisioning the VMs. It takes about 10 minutes to boot and provision the three VMs on the CI.

The advantage of this PR is that we give the developer feedback sooner if there's something wrong with the code. We also avoid wasting time and keeping three CI nodes busy for 10 minutes or more.